toolkit/tools/internal/userutils/userutils.go

```diff
@@ -17,19 +20,25 @@ const (
 	RootUser          = "root"
 	RootHomeDir       = "/root"
 	UserHomeDirPrefix = "/home"
+
+	ShadowFile = "/etc/shadow"
 )
 
 func HashPassword(password string) (string, error) {
 	const postfixLength = 12
 
+	if password == "" {
+		return "", nil
+	}
+
 	salt, err := randomization.RandomString(postfixLength, randomization.LegalCharactersAlphaNum)
 	if err != nil {
 		return "", fmt.Errorf("failed to generate salt for hashed password:\n%w", err)
 	}
 
 	// Generate hashed password based on salt value provided.
 	// -6 option indicates to use the SHA256/SHA512 algorithm
-	stdout, _, err := shell.Execute("openssl", "passwd", "-6", "-salt", salt, password)
+	stdout, _, err := shell.ExecuteWithStdin(password, "openssl", "passwd", "-6", "-salt", salt, "-stdin")
```

Comment:
curiousm, how does the used hash algorithm (-6/sha512) get propagated into the /etc/shadow file entry?

Reply:
The string generated contains the hashing algorithm used.